safenode/src/network/cmd.rs

```diff
@@ -58,6 +77,38 @@ pub struct SwarmLocalState {
 impl SwarmDriver {
     pub(crate) fn handle_cmd(&mut self, cmd: SwarmCmd) -> Result<(), Error> {
         match cmd {
+            SwarmCmd::GetData { key, sender } => {
+                match self.swarm.behaviour_mut().kademlia.store_mut().get(&key) {
```

Comment:
Related to the subsequent comment.
If we're going with just the PUT/GET flow, it might be better to refactor the `GET` flow here. As, calling `kademlia.get_record(key)` starts an iterative query through the network till it finds the data. And I think we can remove `get_closest_peers` that happens prior to a call to this (via `send_to_closest_peers`). 

The only difference between both of the approaches is that, the iterative `get_record()` call short circuits when it finds the first node that holds the data and the data is returned from that node. Whereas, the current flow queries the network till it finds all the `closest_peers`, then request from every single one of them.

Reply:
> The only difference between both of the approaches is that, the iterative get_record() call short circuits when it finds the first node that holds the data and the data is returned from that node.

I think this is key, and where we need to decide what behaviour we want. 

I was still aiming to get data from the supposed holders eg. It may be we dont need to... but may be we do?